src/status_im/extensions/ethereum.cljs

```diff
@@ -234,19 +234,47 @@
                         :on-failure on-failure}]
      (wrap-with-resolution db json-rpc-args :to execute-send-transaction))))
 
-(defn- parse-log [{:keys [address transactionHash blockHash transactionIndex topics blockNumber logIndex removed data]}]
-  (merge {:data    data
+(defn- event-topic-enc [event params]
+  (let [eventid (str event "(" (string/join "," params) ")")]
+    (abi-spec/sha3 eventid)))
+
+; Return a vector without elements of Y
+(defn diff-vectors [X Y]
+  (reduce #(remove (fn [x] (= x %2)) %1) X Y))
+
+; Return a map with all data params decoded
+(defn- decode-data [data events-hints first-topic indexed]
+  (let [index-event (string/index-of (mapv #(event-topic-enc (% :event) (% :params)) events-hints) first-topic)
+        event-hint (get events-hints index-event)
+        params-hint (diff-vectors (event-hint :params) indexed) ; Exclude indexed params from decode in data, these are decoded in topics
+        names-hint (mapv keyword (event-hint :names))
+        data-values (mapv (partial apply str) (partition 64 (string/replace-first data #"0x" "")))]
+    (zipmap names-hint (mapv abi-spec/hex-to-value data-values params-hint))))
+
+; This assumes that Topic 0 is filtered and return a map with all topics decoded
+(defn- decode-topics [topics indexed]
+  (zipmap [:topic1 :topic2 :topic3] (mapv abi-spec/hex-to-value topics indexed)))
+
+; Return the vector of indexed types from the specified event (Topic0) from a vector of event hints given by the user
+(defn- get-indexed [hints first-topic]
+  ((get hints (string/index-of (mapv #(event-topic-enc (% :event) (% :params)) hints) first-topic)) :indexed))
```

Comment:
But here the argument is a vector, not a string? Better to keep `.indexOf` then.

Reply:
Ohh my bad! you are right, fixed it in the new commit. Thanks @jeluard